File: src/main/java/io/snyk/gradle/plugin/SnykExtension.java
package io.snyk.gradle.plugin;

import org.gradle.api.GradleException;

import java.util.Arrays;

public class SnykExtension {

    String arguments;
    String api;
    String severity;
    boolean autoDownload = true;
    boolean autoUpdate = false;

    private static final String[] severities = new String[]{"high", "medium", "low"};

    public void setArguments(String arguments) {
        this.arguments = arguments;
    }

    public void setApi(String api) {
        this.api = api;
    }

    public void setSeverity(String severity) {
        if (Arrays.stream(severities).anyMatch(severity::equals)) {
            this.severity = severity;
        } else {
            throw new GradleException("Severity should either high|medium|low or empty");
        }
    }

    public void setAutoDownload(boolean autoDownload) {
        this.autoDownload = autoDownload;
    }

    public void setAutoUpdate(boolean autoUpdate) {
        this.autoUpdate = autoUpdate;
    }
}

File: src/main/kotlin/io/snyk/gradle/plugin/kt/SnykExtension.kt
package io.snyk.gradle.plugin.kt

import org.gradle.api.file.DirectoryProperty
import org.gradle.api.provider.ListProperty
import org.gradle.api.provider.Property

abstract class SnykExtension {
  abstract val apiKey: Property<String>

  abstract val defaultArguments: ListProperty<String>
  abstract val defaultSeverity: Property<Severity>

  abstract val cliAutoUpdateEnabled: Property<Boolean>
  abstract val cliDownloadDir: DirectoryProperty

  abstract val cliVersion: Property<String>
  abstract val cliFilename: Property<String>
  abstract val cliSourceUri: Property<String>

  enum class Severity(
    val cliArg: String
  ) {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high"),
    CRITICAL("critical"),
  }
}
